browser/resources/settings/email_aliases_page/email_aliases.tsx

```diff
@@ -0,0 +1,449 @@
+    </div>
+  </S.Col>
+)
+
+const MainEmailEntryForm = ({viewState, mainEmail, onEmailSubmitted, restart} : {viewState:ViewState, mainEmail: string, onEmailSubmitted: Function, restart: Function}) => (
+  <S.Card id='main-email-entry-form'>
+    <S.SignupRow>
+      <BraveIcon style={{marginTop: '1em'}}/>
+      {viewState.mode === ViewMode.SignUp ? (<BeforeSendingEmailForm initEmail={mainEmail} onSubmit={onEmailSubmitted}/>) : (<AfterSendingEmailMessage mainEmail={mainEmail} tryAgain={restart}/>)}
+    </S.SignupRow>
+  </S.Card>
+)
+
+const MainView = ({
+  viewState, mainEmail, onLogout, aliasesState, setViewState,
+  mappingService, onListChange
+}: {
+  viewState: ViewState,
+  mainEmail: string,
+  onLogout: Function,
+  aliasesState: Alias[],
+  setViewState: Function,
+  mappingService: MappingService,
+  onListChange: Function
+}) => (
+  (viewState.mode === ViewMode.Startup ?
+    (<S.Row style={{margin: '1em', flexGrow: 1, justifyContent: 'center', alignItems: 'center'}}><Icon name='loading-spinner' />
+      <h3 style={{margin: '0.25em'}}>{getLocale('emailAliasesConnectingToBraveAccount')}</h3>
+     </S.Row>) :
+    (<span>
+      <MainEmailDisplay onLogout={onLogout} email={mainEmail} />
+      <AliasList aliases={aliasesState} onViewChange={setViewState}
+        mappingService={mappingService}
+        onListChange={onListChange}></AliasList>
+    </span>))
+)
+
+export const ManagePage = ({ mappingService }:
+  {
+    mappingService: MappingService
+  }) => {
+  const [viewState, setViewState] = React.useState<ViewState>({ mode: ViewMode.Startup })
+  const [mainEmail, setMainEmail] = React.useState<string>('')
+  const mode = viewState.mode
+  const [aliasesState, setAliasesState] = React.useState<Alias[]>([]);
+  const onEmailChange = async () => {
+    const email = await mappingService.getAccountEmail()
+    setMainEmail(email ?? '')
+    setViewState({ mode: email ? ViewMode.Main : ViewMode.SignUp })
+  }
+  const onListChange = async () => {
+    const aliases = await mappingService.getAliases()
+    setAliasesState(aliases)
+  }
+  const onMainEmailSubmitted = async (email: string) => {
+    setMainEmail(email)
+    await mappingService.requestAccount(email)
+    setViewState({ mode: ViewMode.AwaitingAuthorization })
+    const accountReady = await mappingService.onAccountReady()
+    if (accountReady) {
+      setViewState({ mode: ViewMode.Main })
+      await onListChange()
+    }
+  }
+  const onLogout = () => {
+    mappingService.logout()
+    setViewState({ mode: ViewMode.SignUp })
+  }
+  const restart = async () => {
+    await mappingService.cancelAccountRequest()
+    setViewState({ mode: ViewMode.SignUp })
+  }
+  const returnToMain = () => {
+    setViewState({ mode: ViewMode.Main })
+    onListChange()
+  }
+  React.useEffect(() => {
+    onEmailChange();
```

Comment:
lets be consistent about how we name these. Either all or none should have an `on` prefix

Reply:
fixed